openwillis/measures/audio/util/transcribe_util.py

```diff
@@ -8,26 +8,138 @@
+
+    Parameters:
+    ...........
+    data: json
+        Speech transcription response
+
+    Returns:
+    ...........
+    content_dict : dict
+        A dictionary where the keys are speaker labels and the values are the corresponding content spoken by each
+
+    ------------------------------------------------------------------------------------------------------
+    """
+    content_dict = {}
+
+    if 'results' in data:           # Check if 'results' key is present
+        item_data = data['results']['items']
+
+        item_spk_0 = [item for item in item_data if item.get('speaker_label', '') == 'speaker0']
+        item_spk_1 = [item for item in item_data if item.get('speaker_label', '') == 'speaker1']
+        
+        spk_0_text = [item['alternatives'][0]['content'] for item in item_spk_0 if 'alternatives' in item]
+        spk_1_text = [item['alternatives'][0]['content'] for item in item_spk_1 if 'alternatives' in item]
+
+        content_dict['speaker0'] = " ".join(spk_0_text)
+        content_dict['speaker1'] = " ".join(spk_1_text)
+
+    return content_dict
+
+def get_clinical_lables(scale, measures, content_dict, json_response):
+    """
+    ------------------------------------------------------------------------------------------------------
+
+    Replaces speaker labels in a JSON response based on clinical measures.
+
+    Parameters:
+    ...........
+    scale : str
+        Clinical scale
+    measures : object
+        A configuration object.
+    content_dict: dict
+        A dictionary containing speaker-based content.
+    json_response: json
+        Speech transcription response
+
+    Returns:
+    ...........
+    json_response : json
+        The modified JSON response with replaced speaker labels.
+
+    ------------------------------------------------------------------------------------------------------
+    """
+    #Check if content is available for all the speaker
+    if content_dict and content_dict['speaker0'] and content_dict['speaker1']:
+        if scale.lower() not in measures['scale'].strip("[]").replace(" ", "").split(","):
+            return json_response
+
+        spk1_score = sutil.match_transcript(measures, content_dict['speaker0'])
+        spk2_score = sutil.match_transcript(measures, content_dict['speaker1'])
+
+        if spk1_score > spk2_score:
+            json_response = replace_speaker_labels(json_response, ['speaker0', 'speaker1'], ['clinician', 'participant'])
+
+        else:
+            json_response = replace_speaker_labels(json_response, ['speaker0', 'speaker1'], ['participant', 'clinician'])
+
+    return json_response
+
+def transcribe_audio(s3uri, input_param):
     """
     ------------------------------------------------------------------------------------------------------
 
     Transcribe an audio file using AWS Transcribe.
 
     Parameters:
     ...........
-    s3uri : str
```

Comment:
`s3uri` is still used, so dont remove from docstring

Reply:
updated